[Anjali Agarwal] iP

agarwal-anjali opened the pull request:
### Alpha
_“Time isn’t the main thing. It’s the only thing.”_ - Miles Davis [(source)](https://quotefancy.com/miles-davis-quotes)

Never miss a deadline again! 
**Alpha** is here to help you keep a track of your tasks! 🫡

It's 
- ~EASY~ SUPER EASY to learn
- ~FUN~ SUPER FUN to use

To begin with, all you need to do is:

1. Download it from [here](https://github.com/agarwal-anjali/ip/releases/download/A-Jar/alpha.jar) and double-click on it.
2. Here you go, Alpha is now all yours and that too for **free**!
3. Just enter **_help_** to learn about all commands and their format.
4. That's it, you can now start managing your tasks!

It provides a range of features:

- [x] Add tasks: todos, events and deadlines.
- [x] Mark and unmark tasks as done/pending.
- [x] Find all tasks that contain a specific keyword.
- [x] Delete tasks.
- [x] View the list of tasks.
- [x] Get help.

If you are a Java enthusiast and would like to know about the range of colors Alpha uses to make its UI attractive then, here's the `ANSI_CODE` for all the colors:

    /** Ansi code to add white font color to print statements */
    public final String ANSI_WHITE = "\u001B[37m";

    /** Ansi code to add red font color to print statements */
    public final String ANSI_RED = "\u001B[31m";

    /** Ansi code to add bold blue font color to print statements */
    public final String ANSI_BLUE_BOLD = "\033[1;36m";

    /** Ansi code to add blue font color to print statements */
    public final String ANSI_BLUE = "\u001B[36m";

    /** Ansi code to reset default font color in print statements */
    public final String ANSI_RESET = "\u001B[0m";

Commit:
Add Gradle support

agarwal-anjali committed:
Added welcome message

agarwal-anjali committed:
Formatted and customized display

agarwal-anjali committed:
Implemented a feature to add and display tasks

agarwal-anjali committed:
Implemented a feature to mark and unmark tasks

agarwal-anjali committed:
Created Task class

agarwal-anjali committed:
Implemented a feature to track the type of tasks

agarwal-anjali committed:
Added automated text UI testing files

agarwal-anjali committed:
Added AlphaException file to handle incorrect input errors

agarwal-anjali committed:
Changed list of tasks name from todo to tasks

agarwal-anjali committed:
Implemented a feature to delete tasks

agarwal-anjali committed:
Add a feature to save and retrieve tasks from hard-disk

agarwal-anjali committed:
Format date and time

agarwal-anjali committed:
Merge branch 'branch-Level-7'

agarwal-anjali committed:
Merge branch 'branch-Level-8'

Echomo-Xinyu commented on src/main/java/Alpha.java:
```diff
@@ -0,0 +1,146 @@
+        System.out.println("\n-----------------\n" + ANSI_BLUE + "Hello, I'm ALPHA!" + ANSI_RESET + "\n-----------------");
+    }
+
+    private static void enterMessage(Scanner in) throws AlphaException, IOException {
```
`enterMessage` seems to suggest a function that enters a message into something. Do you think `readMessage` might be a better name as it reflects the action from the chatbot's point of view instead of user's point of view?

Echomo-Xinyu commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+        super(description, taskType);
+        this.deadline = deadline;
+    }
+    public String getDeadline(){
```
Do you think `getDeadlineTimeString()` can be a better time as we would expect a `String` instead of a `Deadline` object?

Similar questions apply for `Event` and `Todo` class.

Echomo-Xinyu commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,30 @@
+    protected String description;
+    protected boolean status;
+
+    protected String taskType;
```
I think an empty line should be left between the `class attributes` and the `constructor`?

Echomo-Xinyu commented on src/main/java/Alpha.java:
```diff
@@ -0,0 +1,146 @@
+import java.io.FileNotFoundException;
```
Missing package header as required.

Echomo-Xinyu reviewed (commented):
It can be seen that much effort has been put into writing this work and most of the function namings and variable namings are following the required conventions.

Considering the progress of the writer so far (till Level-8 as declared in the tag), he or she might need to put in more time into writing this project.

But still, good job for the decent code quality!

agarwal-anjali committed:
Use more OOP

agarwal-anjali committed:
Organize into packages

agarwal-anjali committed:
Reorganize packages

agarwal-anjali committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

agarwal-anjali committed:
Set up gradle and automate

Hongyi6328 reviewed (commented):
Overall, good attempt! A lot of requirements fulfilled.
Just some minor suggestions here.
Would you consider adding more header comments before non-private classes and methods, so that future reviewers can understand your project better?
And don't forget to add a PR description according to the link here! [link](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html)

agarwal-anjali committed:
Add JUnit Tests and an additional help feature

agarwal-anjali committed:
Organize test files into packages

agarwal-anjali committed:
Create a JAR file

agarwal-anjali committed:
Add JavaDoc

agarwal-anjali committed:
Follow the coding standard

agarwal-anjali committed:
Implement a feature to find tasks using keywords

agarwal-anjali committed:
Merge branch 'branch-A-JavaDoc'

agarwal-anjali committed:
Merge branch 'branch-A-CodingStandard'

agarwal-anjali committed:
Merge branch 'branch-Level-9'

agarwal-anjali committed:
Use checkstyle

agarwal-anjali committed:
Add GUI

agarwal-anjali committed:
Merge branch 'branch-Level-10'

agarwal-anjali committed:
Remove extra image files

agarwal-anjali committed:
Removed unwanted files

agarwal-anjali committed:
Add assertions

agarwal-anjali committed:
Improve code quality

agarwal-anjali committed:
Create a method to generate display messages

The execute method in each command class before returning the execution
message passes it to the returnTask method of the UI class which returns
back the message without making any modifications to it.

Creating a method in the UI class to generate the execution message
based on the type of the command being executed.

This improves the code quality by removing redundant method calls and
avoiding code repitition.

agarwal-anjali committed:
Add more assertions

In rewriteFile function there is a redundant else if check to check
whether the task t is an instance of Deadline or not.

Replacing else if check with else as if the task is not an instance of
Todo or Event then it is compelled to be an instance of Deadline.

Add an assertion within the else block to assert that task t is an
instance of Deadline.

agarwal-anjali committed:
Merge pull request #6 from agarwal-anjali/branch-A-Assertions

Branch a assertions

agarwal-anjali committed:
Merge branch 'master' into branch-A-CodeQuality

agarwal-anjali committed:
Merge pull request #7 from agarwal-anjali/branch-A-CodeQuality

Branch a code quality

agarwal-anjali committed:
Fix code style errors and bugs

agarwal-anjali committed:
Set up CI

agarwal-anjali committed:
Add a feature to tag tasks

agarwal-anjali committed:
Add a feature to find tasks with a tag

agarwal-anjali committed:
Modify help feature

agarwal-anjali committed:
Merge branch 'branch-BCD-Extensions'

agarwal-anjali committed:
Fix bugs in test files

agarwal-anjali committed:
Make GUI better

agarwal-anjali committed:
Merge branch 'branch-A-BetterGui'

agarwal-anjali committed:
Give a unique personality

agarwal-anjali committed:
Handle more errors

agarwal-anjali committed:
Set product name as title

agarwal-anjali committed:
Add a User Guide

agarwal-anjali committed:
Modify user guide

agarwal-anjali committed:
Update jar file link in user guide
